src/Serializer.php

```diff
@@ -61,6 +79,18 @@ public function serialize(\Closure $closure)
      */
     public function unserialize($serialized)
     {
+        if ($this->signingKey) {
+            if ($serialized[0] !== '%') {
+                throw new ClosureUnserializationException('You cannot verify '
+                    . 'the signature of an unsigned serialized closure.'
+                );
+            }
+            $signature = base64_decode(substr($serialized, 1, 44));
+            $serialized = substr($serialized, 45);
+
+            $this->verifySignature($signature, $serialized);
+        }
+
         /** @var SerializableClosure $unserialized */
         $unserialized = unserialize($serialized);
```

Comment:
You may want to check for `%` here if signing key is not set (and hence strip off the token). As if you don't verify a signed serialization will not be unserializable otherwise.